PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/filesystem/ADLSGen2FileSystem.java

```diff
@@ -59,16 +59,29 @@ public String getProtocol() {
         return myProtocol;
     }
 
-    public URI getRoot() {return root;}
-
     @NotNull
     public VirtualFile[] listFiles(AdlsGen2VirtualFile vf) {
         List<AdlsGen2VirtualFile> childrenList = new ArrayList<>();
         if (vf.isDirectory()) {
-            childrenList = this.op.list(this.restApiRoot, this.op.getDirectoryParam(vf.getUri()))
-                    .map(path -> new AdlsGen2VirtualFile(this.root.resolve(path.getName()),
-                            path.isDirectory(), this))
+            // sample rootUrl: https://accountName.dfs.core.windows.net/fileSystem
+            URI rootUrl = this.rootPathUri.getRoot().getUrl();
```

Comment:
URI or URL?

Reply:
Not sure what this is about, will sync with you offline.

---
In reply to: [362135752](https://github.com/microsoft/azure-tools-for-java/pull/3885#discussion_r362135752) [](ancestors = 362135752)